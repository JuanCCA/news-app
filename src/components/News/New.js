import React from 'react'
import Card from 'react-bootstrap/Card'
import {archiveNew} from '../../helpers/News/archiveNew'
import {formatDate} from '../../helpers/Utils/dateFormater'
import Button from 'react-bootstrap/Button'


const New = ({title,description,date,content,author,archiveDate,_id, setNews}) => {


    const formatedDate = formatDate(date)

    // Archive a New, adding an archiveDate to this object in DB
    const handlerArchive = async() => {
        const resp = await archiveNew(_id)
        if(resp === 200){
            handlerDeleteFromList()
        }
    }

    // Delete this item from the list 
    const handlerDeleteFromList = () => {
        setNews( (news) => {
            console.log(news)
            return news.filter( doc => doc._id !== _id)
        })
    }

    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Date: {formatedDate} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col" >
                    <Button onClick={handlerArchive}
                            variant="dark">
                        <img 
                        src="./eye-off.png"
                        width="25"
                        height="25"
                        className="d-inline-block align-top"   
                        alt=""                 
                        >
                    </img> Archive
                    </Button>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}


export default New
