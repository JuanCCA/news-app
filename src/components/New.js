import React from 'react'
import Card from 'react-bootstrap/Card'
import {archiveNew} from '../helpers/News/archiveNew'
import {formatDate} from '../helpers/Utils/dateFormater'


const New = ({title,description,date,content,author,archiveDate,_id, setNews}) => {


    const formatedDate = formatDate(date)

    // Archive a New, adding an archiveDate to this object in DB
    const handlerArchive = async() => {
        const resp = await archiveNew(_id)
        if(resp === 200){
            handlerDeleteFromList()
        }
    }

    // Delete this item from the list 
    const handlerDeleteFromList = () => {
        setNews( (news) => {
            console.log(news)
            return news.filter( doc => doc._id !== _id)
        })
    }

    return (
        <Card style={{ width: '90%' }} className="my-3">
        <Card.Body>
            <Card.Title>{title}</Card.Title>
            <Card.Subtitle className="mb-2 text-muted">{description}</Card.Subtitle>
            <Card.Text>{content}</Card.Text>
            <Card.Footer >

            <div className="row">
                <div className="col">
                    Date: {formatedDate} 
                </div>
                <div className="col">
                    Author: {author}
                </div>
                <div className="col">
                    <button onClick={handlerArchive}>
                        <img 
                        src="./eye-off.png"
                        width="30"
                        height="30"   
                        alt=""                 
                        >
                    </img> Archive
                    </button>
                </div>
            </div>
                
            </Card.Footer>
        </Card.Body>
        </Card>
    )
}


export default New
