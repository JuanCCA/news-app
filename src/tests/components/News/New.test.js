import { shallow } from 'enzyme'
import React from 'react'
import New from '../../../components/News/New'



describe('Testing component <New />', () => {
    
    const args = {
        _id:"28929837466959830853",
        title: "TITLE 1",
        description: "Description of the new",
        content: "content of the new",
        date: "2021-09-09T22:00:00.000+00:00",
        author: "Juan Carlos"
    }

    const setNews = jest.fn()

    let wrapper = shallow(<New key={args._id}  {...args} setNews={setNews}/>)

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<New key={args._id}  {...args} setNews={setNews}/>);
    });


    test('should create the component correctly with all of arguments needed', () => {
        
        expect(wrapper).toMatchSnapshot()


    })





})
