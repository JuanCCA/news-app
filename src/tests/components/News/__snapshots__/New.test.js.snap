// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <New /> should create the component correctly with all of arguments needed 1`] = `
<Card
  body={false}
  className="my-3"
  style={
    Object {
      "width": "90%",
    }
  }
>
  <CardBody>
    <CardTitle>
      TITLE 1
    </CardTitle>
    <CardSubtitle
      className="mb-2 text-muted"
    >
      Description of the new
    </CardSubtitle>
    <CardText>
      content of the new
    </CardText>
    <CardFooter>
      <div
        className="row"
      >
        <div
          className="col"
        >
          Date: 
          2021/09/10  00:00
        </div>
        <div
          className="col"
        >
          Author: 
          Juan Carlos
        </div>
        <div
          className="col"
        >
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="dark"
          >
            <img
              alt=""
              className="d-inline-block align-top"
              height="25"
              src="./eye-off.png"
              width="25"
            />
             Archive
          </Button>
        </div>
      </div>
    </CardFooter>
  </CardBody>
</Card>
`;
